demomultistoreform/src/Grid/Query/ContentBlockQueryBuilder.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoMultistoreForm\Grid\Query;
+
+use Doctrine\DBAL\Connection;
+use PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use \Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Adapter\Shop\Context;
+
+final class ContentBlockQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var Context
+     */
+    private $shopContext;
+
+    /**
+     * ContentBlockQueryBuilder constructor.
+     *
+     * @param Connection $connection
+     * @param $dbPrefix
+     * @param Context $shopContext
+     */
+    public function __construct(Connection $connection, $dbPrefix, Context $shopContext)
+    {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->shopContext = $shopContext;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return QueryBuilder
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria): QueryBuilder
+    {
+        $qb = $this->getBaseQuery();
+        $qb->select('cb.id_content_block, cb.title, cb.description, cb.enable');
+
+        if ($this->shopContext->isSingleShopContext() || $this->shopContext->isGroupShopContext()) {
```

Comment:
```suggestion
        if (!$this->shopContext->isAllShopContext()) {
```

this should coming from `MultistoreContextCheckerInterface`